huaweicloud/services/workspace/data_source_huaweicloud_workspace_desktop_tags_filter.go

```diff
@@ -0,0 +1,303 @@
+package workspace
+
+import (
+	"context"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+// @API Workspace POST /v2/{project_id}/desktops/resource_instances/action
+func DataSourceDesktopTagsFilter() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceDesktopTagsFilterRead,
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: `The region where the desktop tags are located.`,
+			},
+			"without_any_tag": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: `Specifies whether to query resources without any tag.`,
+			},
+			"matches": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Elem:        matchSchema(),
```

Comment:
```suggestion
				Elem:        desktopTagsQueriedMatchFilterSchema(),
```
desktopTagsQueriedXXXFilter

Reply:
done